monix-eval/shared/src/test/scala/monix/eval/TaskGatherNSuite.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.eval
+
+import monix.execution.exceptions.DummyException
+import monix.execution.internal.Platform
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object TaskGatherNSuite extends BaseTestSuite {
+
+  test("Task.gatherN should execute in parallel for async tasks") { implicit s =>
+    val seq = Seq(
+      Task.evalAsync(1).delayExecution(1.second),
+      Task.evalAsync(2).delayExecution(2.seconds),
+      Task.evalAsync(3).delayExecution(3.seconds)
+    )
+    val f = Task.gatherN(1)(seq).runToFuture
+
+    s.tick()
+    assertEquals(f.value, None)
+    s.tick(1.second)
+    assertEquals(f.value, None)
+    s.tick(2.seconds)
+    assertEquals(f.value, None)
+    s.tick(3.seconds)
+    assertEquals(f.value, Some(Success(Seq(1, 2, 3))))
+  }
+
+  test("Task.gatherN should onError if one of the tasks terminates in error") { implicit s =>
+    val ex = DummyException("dummy")
+    val seq = Seq(
+      Task.evalAsync(3).delayExecution(2.seconds),
+      Task.evalAsync(2).delayExecution(1.second),
+      Task.evalAsync(throw ex).delayExecution(2.seconds),
+      Task.evalAsync(3).delayExecution(1.seconds)
+    )
+
+    val f = Task.gatherN(2)(seq).runToFuture
+
+    s.tick()
+    assertEquals(f.value, None)
+    s.tick(2.seconds)
+    assertEquals(f.value, None)
+    s.tick(2.seconds)
+    assertEquals(f.value, Some(Failure(ex)))
+  }
+
+  test("Task.gatherN should be canceled") { implicit s =>
+    val seq = Seq(
+      Task.evalAsync(1).delayExecution(2.seconds),
+      Task.evalAsync(2).delayExecution(1.second),
+      Task.evalAsync(3).delayExecution(3.seconds))
+    val f = Task.gatherN(1)(seq).runToFuture
+
+    s.tick()
+    assertEquals(f.value, None)
+    s.tick(1.second)
+    assertEquals(f.value, None)
+
+    f.cancel()
+    s.tick(1.second)
```

Comment:
That's actually not checking anything. The `f` can only complete after 6 seconds. I recommend having an effect in `doOnCancel` on 1st Task which you can actually check and then effect on 2nd Task which you can see _not_ happening even after ticking for `1.day`